processing/src/main/java/org/apache/druid/data/input/impl/CloudObjectSplitWidget.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Helper used by {@link CloudObjectInputSource} to implement {@link SplittableInputSource#createSplits}.
+ */
+public interface CloudObjectSplitWidget
+{
+  long UNKNOWN_SIZE = -1;
```

Comment:
Looks like we do have an unknown size. Yet, in Google, we use the max long value when the size is unknown. Do we have a consistent story?

Reply:
Ah, the Google code snippet we were talking about in another thread doesn't operate in a context where this `UNKNOWN_SIZE` would be appropriate. (That other one is fed into a SplitHintSpec, which don't expect unknown sizes.)